packages/build-tools/src/steps/functions/repack.ts

```diff
@@ -0,0 +1,277 @@
+                tmpDir,
+                tmpFiles,
+              }),
+              logger: repackLogger,
+              spawnAsync: repackSpawnAsync,
+              verbose,
+              env,
+            });
+            break;
+        }
+      } finally {
+        await Promise.all(tmpFiles.map((file) => fs.promises.rm(file, { force: true })));
+      }
+
+      stepsCtx.logger.info(`Repacked the app to ${outputPath}`);
+      outputs.output_path.set(outputPath);
+    },
+  });
+}
+
+/**
+ * Creates a Bunyan logger adapter for repack logger
+ */
+export function createBunyanLoggerAdapter(logger: bunyan): Logger {
+  const timerMap: Record<string, number> = {};
+
+  return {
+    debug(...message: any[]): void {
+      logger.debug(message.join(''));
+    },
+    info(...message: any[]): void {
+      logger.info(message.join(''));
+    },
+    warn(...message: any[]): void {
+      logger.warn(message.join(''));
+    },
+    error(...message: any[]): void {
+      logger.error(message.join(''));
+    },
+    time(label: string): void {
+      timerMap[label] = Date.now();
+    },
+    timeEnd(label: string): void {
+      if (timerMap[label] == null) {
+        logger.warn(`Timer '${label}' does not exist`);
+        return;
+      }
+      const duration = Date.now() - timerMap[label];
+      logger.info(`${label}: ${duration} ms`);
+      delete timerMap[label];
+    },
+  };
+}
+
+/**
+ * Creates `@expo/steps` based spawnAsync for repack.
+ */
+function createSpawnAsyncStepAdapter({
+  verbose,
+  logger,
+}: {
+  verbose: boolean;
+  logger: bunyan;
+}): SpawnProcessAsync {
+  return function repackSpawnAsync(
+    command: string,
+    args: string[],
+    options?: SpawnProcessOptions
+  ): SpawnProcessPromise<SpawnProcessResult> {
+    return spawnAsync(command, args, {
+      ...options,
+      ...(verbose ? { logger, stdio: 'pipe' } : { logger: undefined }),
+    });
+  };
+}
+
+/**
+ * A helper function to infer the platform job type based on the provided platform.
+ */
+function getPlatformJob<T extends Job>(job: T, platform: 'android'): Android.Job;
+function getPlatformJob<T extends Job>(job: T, platform: 'ios'): Ios.Job;
+function getPlatformJob<T extends Job>(job: T, _: string): T {
+  return job;
+}
+
+/**
+ * Resolves Android signing options from the job secrets.
+ */
+export async function resolveAndroidSigningOptionsAsync({
+  job,
+  tmpDir,
+  tmpFiles,
+}: {
+  job: Job;
+  tmpDir: string;
+  tmpFiles: string[];
+}): Promise<AndroidSigningOptions | undefined> {
+  const androidJob = getPlatformJob(job, 'android');
+  const buildCredentials = androidJob.secrets?.buildCredentials;
+  if (buildCredentials?.keystore.dataBase64 == null) {
+    return undefined;
+  }
+  const keyStorePath = path.join(tmpDir, `keystore-${randomUUID()}`);
+  await fs.promises.writeFile(
+    keyStorePath,
+    Buffer.from(buildCredentials.keystore.dataBase64, 'base64')
+  );
+  tmpFiles.push(keyStorePath);
```

Comment:
Passing mutating arguments feels quite weird, at least I don't see it too often. If we leave this functions here, could we instead add `tmpFiles` to output of the function? Or maybe, since `keyStorePath` is the only file that will be created in the process, we might, instead of wrapping both repack calls in one huge `try-catch`, do

```
switch ... {
  case ...:
    let keyStorePath;
    try {
      const credentials = ...;
      keyStorePath = credentials.keyStorePath;
      repack(...);
    } finally {
      if keyStorePath ( rm )
    }
...
}
```

It's definitely more verbose, but I feel like it raises less questions

Reply:
addressed in latest commit